homeassistant/components/sensor/sabnzbd.py

```diff
@@ -4,216 +4,72 @@
-    await sab_api.refresh_data()
-
-
-def request_configuration(host, name, hass, config, async_add_devices,
-                          sab_api):
-    """Request configuration steps from the user."""
-    configurator = hass.components.configurator
-    # We got an error if this method is called while we are configuring
-    if host in _CONFIGURING:
-        configurator.notify_errors(_CONFIGURING[host],
-                                   'Failed to register, please try again.')
-
-        return
-
-    @asyncio.coroutine
-    def async_configuration_callback(data):
-        """Handle configuration changes."""
-        api_key = data.get('api_key')
-        if (yield from async_check_sabnzbd(sab_api, host, api_key)):
-            setup_sabnzbd(host, api_key, name, config,
-                          async_add_devices, sab_api)
-
-            def success():
-                """Set up was successful."""
-                conf = load_json(hass.config.path(CONFIG_FILE))
-                conf[host] = {'api_key': api_key}
-                save_json(hass.config.path(CONFIG_FILE), conf)
-                req_config = _CONFIGURING.pop(host)
-                configurator.async_request_done(req_config)
-
-            hass.async_add_job(success)
-
-    _CONFIGURING[host] = configurator.async_request_config(
-        DEFAULT_NAME,
-        async_configuration_callback,
-        description='Enter the API Key',
-        submit_caption='Confirm',
-        fields=[{'id': 'api_key', 'name': 'API Key', 'type': ''}]
-    )
-
-
-@asyncio.coroutine
-def async_setup_platform(hass, config, async_add_devices, discovery_info=None):
-    """Set up the SABnzbd platform."""
-    from pysabnzbd import SabnzbdApi
-
-    if discovery_info is not None:
-        host = discovery_info.get(CONF_HOST)
-        port = discovery_info.get(CONF_PORT)
-        name = DEFAULT_NAME
-        use_ssl = discovery_info.get('properties', {}).get('https', '0') == '1'
-    else:
-        host = config.get(CONF_HOST)
-        port = config.get(CONF_PORT)
-        name = config.get(CONF_NAME, DEFAULT_NAME)
-        use_ssl = config.get(CONF_SSL)
-
-    api_key = config.get(CONF_API_KEY)
-
-    uri_scheme = 'https://' if use_ssl else 'http://'
-    base_url = "{}{}:{}/".format(uri_scheme, host, port)
-
-    if not api_key:
-        conf = load_json(hass.config.path(CONFIG_FILE))
-        if conf.get(base_url, {}).get('api_key'):
-            api_key = conf[base_url]['api_key']
-
-    if not (yield from async_check_sabnzbd(SabnzbdApi, base_url, api_key)):
-        request_configuration(base_url, name, hass, config,
-                              async_add_devices, SabnzbdApi)
-        return
-
-    setup_sabnzbd(base_url, api_key, name, config,
-                  async_add_devices, SabnzbdApi)
+
+async def async_setup_platform(hass, config, async_add_devices,
+                               discovery_info=None):
+    """Set up the SABnzbd sensors."""
+    sab_api_data = hass.data[DATA_SABNZBD]
```

Comment:
We only set up the sensor platform from the component now, correct? So otherwise we don't want to do anything here. Add a guard clause that checks if discovery_info is None and returns if so.

```py
if discovery_info is None:
    return
...
```

Reply:
Yep, that's correct. Fixed